docs/bc-government-api-guidelines.md

```diff
@@ -0,0 +1,209 @@
+- **_Use gateways and proxies instead of IP whitelists_** – When exposing APIs to the internet, use a secure gateway layer to provide a security control point instead of simply whitelisting inbound Internet Protocol addresses (IPs). When consuming external APIs, route flows through a forward (egress) proxy instead of using IP address whitelisting on the outbound firewall
+- **_Integrate Security Testing_**  **-** automate security testing to validate any new changes to API source code and to ensure robustness of requested changes. Assess the change impact and conduct testing accordingly
+- **_Audit Access to Sensitive Data_** – all API based access to non-public data must be logged and retained for audit purposes.  Logging attributes must include the source system, client identifier and associated timestamp from the target system
+- **_Monitor and Log API Activity_** – track API usage and activity to identify performance bottlenecks, peak usage periods and abnormal access patterns. Use open standards-based logging frameworks such as CEF (Common Event Format) and collect logs in a central repository
+  - When usage limits are exceeded, by default, a 1-hour timeout (block) will be used
+  - X-RATELIMIT-LIMIT and X-RATELIMIT-REMAINING can be inspected in the HTTP response to view current usage
+
+## API Encoding and Metadata
+
+Consistent metadata and encoding ensures that APIs are interoperable across organizations and helps to maintain data consistency. The following practices should be followed when defining your API:
+
+- **_Use Unicode for Encoding_** – Unicode Transformation Format-8 ([UTF-8](http://unicode.org/faq/utf_bom.html#utf8-1)) is the standard encoding type for all text and textual representations of data through APIs. It must be used for all APIs published across the BC Government for both internal and external consumption
+- **_Standardize Datetime Format_** – use the ISO 8601 Standard for datetime representation in all BC Government APIs.  The standard date format is YYYY-MM-DD while timestamp format YYYY—MM-DD HH24:MI:SS.  If other formats are required due to source system limitations, convert it to the standard format within the API
+- **_Support Official Languages_** – ensure the API can return responses in both English and French.   External facing APIs must reply with content in the requested language if the backend data support it. APIs must interpret the ACCEPT-LANGUAGE HTTP header and return the appropriate content. If the header is not set, then content in both languages should be returned
+
+
+## API Documentation
+
+BC Government APIs must be published to the [BC Government API Registry](https://catalogue.data.gov.bc.ca/group/bc-government-api-registry) to be discoverable.  The documentation MVP for BC Government APIs includes everything the API does, including resources, endpoints and methods, parameters, error codes, and example requests and responses:
```

Comment:
This is no longer true and since it is phrased as a _must_ do (rather than most of the other guidelines which are just that), I'd remove this entirely. 

Instead, API providers should be suggested to list their APIs on the [API Directory](https://api.gov.bc.ca/devportal/api-directory), if their API may be accessed by third parties outside the organization (e.g. team or branch) that created them. Docs for sharing APIs on the API Directory - https://developer.gov.bc.ca/docs/default/component/aps-infra-platform-docs/how-to/api-discovery/

Reply:
Hi @rustyjux - I've updated the following based on your comments above:

- removed `API` from all heading sections. So, `API Design` is now `Design` etc.
- I've updated the first part of the `Documentation` section based on your above comments. Please let me know if it looks ok now

Outstanding:
- I didn't know where to put the link to the  https://github.com/bcgov/api-guidelines repo. Please suggest where in the docs it should go.

Reply:
I made a couple additional changes but looks good to go after that.

Re: https://github.com/bcgov/api-guidelines, I don't think we need to reference that on this page, but I do think the managers of the API guidelines (who? @jeff-card ?) need to decide what is the source of truth - ie is it this content now in `bcgov/bcdg` or is it `bcgov/api-guidelines`.

To my eyes, the API guidelines fit right in the BC Developers' Guide (bcdg) and `api-guidelines` can be archived.

Reply:
I think they fit in the BCDG as well, as either a cross reference or as the authoritative source.  I'm fairly confident this repo is widely referenced in documentation, wikis, guides throughout our landscape.

Reply:
@rustyjux - I've committed your changes and the updates are [available on the preview site](https://f5ff48-developer-portal-dev-dprepo-pr-182-f5ff48-dev.apps.silver.devops.gov.bc.ca/docs/default/component/bc-developer-guide/bc-government-api-guidelines/).

@jeff-card - Sorry, I don't think I understand. Do you want the https://github.com/bcgov/api-guidelines repo linked somewhere in this guidelines page?

I will merge this PR after interregnum period is over.